runtime/examples/wasm-program.rs

```diff
@@ -0,0 +1,27 @@
+use rustmatic_wasm::{Program};
```

Comment:
@Michael-F-Bryan 
I'll guess I need a wasm program to test this. What's the best way of organizing this, now that the wasm examples are moved to the temp_folder solution?

Reply:
@Michael-F-Bryan Started looking at this again after a little hiatus.  Took a bit of time to remember where I left this, but seems I just need a wasm program now.

I guess the wasm tests in ```wasm_test``` are generated on the fly, and then deleted again. I'm no sure whether the runtime crate should contain a wasm test (ideally all wasm tests for ci?) or whether this should be placed somewhere else. 

I looked at your wasm_test::Compiler, this seems general enough to be used everywhere we need to compile some wasm?

Reply:
> I looked at your wasm_test::Compiler, this seems general enough to be used everywhere we need to compile some wasm?

Yep. You should be able to pass it some Rust source code and it'll compile everything to `wasm32-unknown-unknown` making sure it gets linked with our standard library.

> I guess the wasm tests in wasm_test are generated on the fly, and then deleted again. I'm no sure whether the runtime crate should contain a wasm test (ideally all wasm tests for ci?) or whether this should be placed somewhere else.

I don't see why not. If we want to use WebAssembly for implementing programs and our runtime is in charge of executing programs and interfacing it with the machine, the runtime code should try to execute WebAssembly code as part of its test suite.

> What's the best way of organizing this, now that the wasm examples are moved to the temp_folder solution?

Can you remember what you meant by this? Normally if I'm writing integration tests for a large library I'll put test fixtures (example data, known inputs, etc.) in a `data/` folder inside my `tests/` folder. If you're crafting programs that are used purely for testing the runtime, a nice place for them would be `runtime/tests/data/`. 

Otherwise if we have example programs people can base their real programs on, we could have a top-level `examples/` directory and wire up CI to run `wasm_test::Compiler` over each to ensure they all compile.

Reply:
>Can you remember what you meant by this? Normally if I'm writing integration tests for a large library I'll put test fixtures (example data, known inputs, etc.) in a data/ folder inside my tests/ folder. If you're crafting programs that are used purely for testing the runtime, a nice place for them would be runtime/tests/data/.

This was actually just what I was looking for, great advice. :+1: 

Reorganized into test directory in f911db9 .